request/src/main/java/com/vimeo/networking2/internal/VimeoApiClientImpl.kt

```diff
@@ -0,0 +1,835 @@
+        val uri = followable.metadata?.interactions?.follow?.uri
+            ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return updateFollow(isFollowing, uri, callback)
+    }
+
+    override fun updateVideoLike(
+        isLiked: Boolean,
+        uri: String,
+        password: String?,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest {
+        val optionsMap = password.asPasswordParameter()
+        return if (isLiked) {
+            vimeoService.put(authHeader, uri, optionsMap).enqueue(callback)
+        } else {
+            vimeoService.delete(authHeader, uri, optionsMap).enqueue(callback)
+        }
+    }
+
+    override fun updateVideoLike(
+        isLiked: Boolean,
+        video: Video,
+        password: String?,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest {
+        val uri = video.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return updateVideoLike(isLiked, uri, password, callback)
+    }
+
+    override fun updateVideoWatchLater(
+        isWatchLater: Boolean,
+        uri: String,
+        password: String?,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest {
+        val optionsMap = password.asPasswordParameter()
+        return if (isWatchLater) {
+            vimeoService.put(authHeader, uri, optionsMap).enqueue(callback)
+        } else {
+            vimeoService.delete(authHeader, uri, optionsMap).enqueue(callback)
+        }
+    }
+
+    override fun updateVideoWatchLater(
+        isWatchLater: Boolean,
+        video: Video,
+        password: String?,
+        callback: VimeoCallback<Unit>
+    ): VimeoRequest {
+        val uri = video.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return updateVideoWatchLater(isWatchLater, uri, password, callback)
+    }
+
+    override fun createComment(
+        uri: String,
+        comment: String,
+        password: String?,
+        callback: VimeoCallback<Comment>
+    ): VimeoRequest {
+        val optionsMap = password.asPasswordParameter()
+        return vimeoService.createComment(
+            authHeader,
+            uri,
+            optionsMap,
+            mapOf(ApiConstants.Parameters.PARAMETER_COMMENT_TEXT_BODY to comment)
+        ).enqueue(callback)
+    }
+
+    override fun createComment(
+        video: Video,
+        comment: String,
+        password: String?,
+        callback: VimeoCallback<Comment>
+    ): VimeoRequest {
+        val uri = video.metadata?.connections?.comments?.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return createComment(uri, comment, password, callback)
+    }
+
+    override fun fetchProductList(
+        fieldFilter: String?,
+        cacheControl: CacheControl?,
+        callback: VimeoCallback<ProductList>
+    ): VimeoRequest {
+        return vimeoService.getProducts(authHeader, fieldFilter, cacheControl).enqueue(callback)
+    }
+
+    override fun fetchProduct(
+        uri: String,
```

Comment:
Why not uniformly validate uri parameters? What happens if this uri is empty for example?